### ISSUE TITLE
MuseScore adds nonexistent rests when importing XML and then thinks it's a corrupt file

### ISSUE THREAD

**Issue by ChrisAm1224** [musescore/MuseScore#22857]
### Issue type

Opening/saving issue

### Steps to reproduce

Open "Brahms 4th Symphony, Andante Moderato Viola m 86 issue with MuseScore.SHY.xml" in 
[MuseScore issues.zip](https://github.com/musescore/MuseScore/files/15333799/MuseScore.issues.zip)
This alert appears: 
![image](https://github.com/musescore/MuseScore/assets/48877076/b4bc9e58-c0c2-4e04-ae1e-b0f93b9c0c71)
Click Open anyway. Display looks like this:
![image](https://github.com/musescore/MuseScore/assets/48877076/5a48dbc0-3c03-4cef-ae5c-b55452273f2c)
MuseScore has added a rest in the end and some tuplets, both of which are not in the original XML, which is supposed to look like this: 
![image](https://github.com/musescore/MuseScore/assets/48877076/0364b453-e11a-46be-8949-34b778493855)
In order to avoid the curruption message, you can open "Brahms 4th Symphony, Andante Moderato Viola m 86 issue with MuseScore fixed.SHY.xml" which contains an additional invisible rest, but why is this necessary? And why are tuplets being added that are not in the original XML?

### MuseScore Version

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.3.0-241231433, revision: github-musescore-musescore-5f36e74

### Regression

No.

### Operating system

Windows 11

**Comment by Jojo-Schmitz** [comment]
Same issue with Mu3, Not with Mu2 or Mu1 though

**Comment by Jojo-Schmitz** [comment]
Might be related to (the root cause of) #22344

**Comment by ChrisAm1224** [comment]
> Might be related to (the root cause of) #22344

not sure though, since 22344 seems to be about export, but this problem is about import

**Comment by Jojo-Schmitz** [comment]
True. But is is about rounding issue due tomusicxml im- and export not yet using `Fraction`, something that got introduced with Mu3

**Comment by ChrisAm1224** [comment]
In the example in this bug (Brahms 4th Symphony, Andante Moderato Viola m 86 issue with MuseScore.SHY.xml), divisions is 24 (i.e., the duration of a quarter note). The file consists of all sixteenth triplets, i.e., duration = 4, (one sixth of a quarter note), no rounding error happening here. The issue seems to be that because of the time modification, MuseScore seems to recognize (correctly) that the notes are triplets. It then tries to group them in groups of three (adding the tuplets that are not there either). In the file, after the alto clef (and the backup tag) there are only two notes in a group (beam), and then another three. MuseScore assumes (incorrectly) that all groups must consist of three notes and then in order to force that, just adds a rest in the end (or that's the best explanation that I can come up with; I haven't checked any source code).

In 22344 what I see is divisions = 160, and 7 septuplet notes with duration 23 each, and 7 * 23 = 161, which could qualify as a "rounding error". There are two ways to fix that
(1) make divisions a multiple of 7
(2) if that overflows (because least common multiple might be too large), make one of the notes (in the middle somewhere) duration 22 so the sum is 160. I've seen other software to the latter.
I see that 22344 has already been solved, I'm curious what the solution will be there.

**Comment by lvinken** [comment]
The MusicXML importer indeed adds rest to incomplete tuplets. This was suggested long time ago by Nicolas Froment. The algorithm is not smart enough to add a rest at the correct location (immediately before the alto clef, as shown in the fixed file). In the current importer design, enhancing the "adds rest to incomplete tuplets" algorithm seems quite difficult. Removing it would be easy, but the impact of that would have to be investigated.

**Comment by ChrisAm1224** [comment]
Notice there are not even any tuplets in the original file, just notes with time modifications to which MuseScore then adds arbitrary tuplet groupings. The measure is from a real-life composition (Brahms 4th symphony slow movement) and there are no tuplets in all the editions that I could find. Triplets are implied because there is a whole run of them. The file I added with an invisible rest ist just to illustrate, but notice MusicXML makes a distinction between time modification and tuplets on purpose.

Good to hear that the solution for 22344 is making divisions a multiple of 7, but as I remarked earlier, if there are e.g., 7-tuplets, 13-tuplets, 17-tuplets in the same file (many numbers that are coprime) the calculation of LCM (least common multiple) might overflow and you will still need a solution as I outlined with not all numbers being the same.

**Comment by ChrisAm1224** [comment]
Here is the place from the Breitkopf edition downloaded from IMSLP:
![image](https://github.com/musescore/MuseScore/assets/48877076/446b4e94-0777-42b3-994c-7d5657b1ac20)

**Comment by rettinghaus** [comment]
There is definitely something wrong with the file and not with MuseScore itself. I tested it with other programs:

**Dorico import**:

![grafik](https://github.com/musescore/MuseScore/assets/7693447/e8845144-196f-4d78-9952-bbc077a4f593)

**Sibelius import**:

![grafik](https://github.com/musescore/MuseScore/assets/7693447/1a5ad285-1b19-4f37-8348-bbb4ecb3852c)

**Comment by ChrisAm1224** [comment]
On the other hand, here is the picture from Finale NotePad: 
![image](https://github.com/musescore/MuseScore/assets/48877076/0d1842fd-7bcc-49d3-9bab-fa60752aa121)
Dorico also adds tuplet marks that are not in the original file and then adds dots that are also not needed. Sibelius too, but it is even stranger: Notes that should clearly be horizontally aligned, because they happen at the same time according to duration (note and backup) and note value plus time modification, are arbitrarily moved to the right or left.

**Comment by ChrisAm1224** [comment]
I just had the idea to add a supports tag (https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/supports/) like this
`<supports element="tuplet" type="yes"/>`
under score-partwise/identification/encoding to indicate the absence of tuplet tags really means there shouldn't be any tuplets. However, that unfortunaly doesn't help either.

**Comment by rettinghaus** [comment]
@ChrisAm1224 If NotePad can handle it, maybe try to re-export it from there?

**Comment by ChrisAm1224** [comment]
did that:
[Brahms 4th Symphony, Andante Moderato Viola m 86 issue with MuseScore.SHY.xml.finale notepad.zip](https://github.com/user-attachments/files/15764511/Brahms.4th.Symphony.Andante.Moderato.Viola.m.86.issue.with.MuseScore.SHY.xml.finale.notepad.zip)
![image](https://github.com/musescore/MuseScore/assets/48877076/0a9d5f6a-27f6-4a38-9518-3bad8c1e7281)
![image](https://github.com/musescore/MuseScore/assets/48877076/64d53d45-f5e4-4007-9ffc-83142c7ae804)
strange though, they add invisible tuplets (no number, no bracket)
`<tuplet bracket="no" number="1" show-number="none" type="start"/>`
can't live without tuplets, can we?
